openedx/core/djangoapps/enrollments/views.py

```diff
@@ -987,3 +995,113 @@ def get_queryset(self):
         return queryset
+
+
+class EnrollmentAllowedView(APIView):
+    """
+    A view that allows the retrieval and creation of enrollment allowed for a given user email and course id.
+    """
+    authentication_classes = (
+        JwtAuthentication,
+    )
+    permission_classes = (IsStaffOrOwner,)
+    throttle_classes = (EnrollmentUserThrottle,)
+    serializer_class = CourseEnrollmentAllowedSerializer
+
+    def get(self, request):
+        """
+        Returns the enrollment alloweds for a given user email.
+
+        **Example Requests**
+
+        GET /api/enrollment/v1/enrollment_allowed?email=user@example.com
+
+        **Parameters**
+
+        - `email` (optional, string, _query_params_)
+
+        **Responses**
+        - 200: Success.
+        - 403: Forbidden, you need to be staff.
+        """
+        user_email = request.query_params.get('email')
+        if not user_email:
+            user_email = request.user.email
+
+        enrollments = CourseEnrollmentAllowed.objects.filter(email=user_email) or []
+        serialized_enrollments = [CourseEnrollmentAllowedSerializer(enrollment).data for enrollment in enrollments]
+
+        return Response(
+            status=status.HTTP_200_OK,
+            data=serialized_enrollments
+        )
+
+    def post(self, request):
+        """
+        Creates an enrollment allowed for a given user email and course id.
+
+        **Example Request**
+
+        POST /api/enrollment/v1/enrollment_allowed
+
+        Example request data:
+        ```
+        {
+            "email": "user@example.com",
+            "course_id": "course-v1:edX+DemoX+Demo_Course",
+            "auto_enroll": true
+        }
+        ```
+
+        **Parameters**
+
+        - `email` (**required**, string, _body_) - defaults to the calling user if not provided.
+
+        - `course_id` (**required**, string, _body_)
+
+        - `auto_enroll` (optional, bool: default=false, _body_)
+
+        **Responses**
+        - 200: Success, enrollment allowed found.
+        - 400: Bad request, missing data.
+        - 403: Forbidden, you need to be staff.
+        - 409: Conflict, enrollment allowed already exists.
+        """
+        email = request.data.get('email')
+        course_id = request.data.get('course_id')
+        auto_enroll = request.data.get('auto_enroll', False)
+
+        if not email:
+            return Response(
+                status=status.HTTP_400_BAD_REQUEST,
+                data={
+                    "message": "Please provide a value for 'email' in the request data."
+                }
+            )
+        elif not course_id:
+            return Response(
+                status=status.HTTP_400_BAD_REQUEST,
+                data={
+                    "message": "Please provide a value for 'course_id' in the request data."
+                }
+            )
+
+        try:
+            enrollment = CourseEnrollmentAllowed.objects.create(
```

Comment:
Even if it is only used twice in the view, `enrollment` is confusing for an CourseEnrollmentAllowed object. Would you agree to call it `enrollment_allowed` or `allowed`?